go/vt/vtgate/engine/unionall.go

```diff
@@ -0,0 +1,72 @@
+package engine
+
+import (
+	"sync"
+
+	"github.com/youtube/vitess/go/sqltypes"
+	"github.com/youtube/vitess/go/vt/concurrency"
+	"github.com/youtube/vitess/go/vt/vterrors"
+	"github.com/youtube/vitess/go/vt/vtgate/queryinfo"
+)
+
+// UnionAll specifies the parameters for a union primitive
+type UnionAll struct {
+	Left, Right Primitive `json:",omitempty"`
+}
+
+// Execute performs an execute
+func (union *UnionAll) Execute(vcursor VCursor, queryConstruct *queryinfo.QueryConstruct, joinvars map[string]interface{}, wantfields bool) (*sqltypes.Result, error) {
+	var lres, rres *sqltypes.Result
+	allErrors := new(concurrency.AllErrorRecorder)
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func(vc *VCursor, qc *queryinfo.QueryConstruct, jv map[string]interface{}, wf bool) {
+		defer wg.Done()
+		var err error
+		lres, err = union.Left.Execute(*vc, qc, jv, wf)
+		allErrors.RecordError(err)
+	}(&vcursor, queryConstruct, joinvars, wantfields)
+	wg.Add(1)
+	go func(vc *VCursor, qc *queryinfo.QueryConstruct, jv map[string]interface{}, wf bool) {
+		defer wg.Done()
+		var err error
+		rres, err = union.Right.Execute(*vc, qc, jv, wf)
+		allErrors.RecordError(err)
+	}(&vcursor, queryConstruct, joinvars, wantfields)
+	wg.Wait()
+	if allErrors.HasErrors() {
+		return nil, allErrors.AggrError(vterrors.Aggregate)
+	}
+	// TODO(acharis): deal with column type mismatch
```

Comment:
If the check is going to be efficient, we can have AppendResult do it. Otherwise, we should make a separate check so that we don't tax the other cases.